multiqc/modules/atropos/atropos.py

```diff
@@ -0,0 +1,1356 @@
+            raise UserWarning
+
+        # Set assets on parent
+        self.atropos_assets(parent)
+
+        # Add to general stats
+        if self.atropos_general_data:
+            self.atropos_general(parent)
+
+        self.atropos_plots(parent)
+
+    def atropos_assets(self, parent):
+        """Add assets to parent."""
+        pass
+
+    def atropos_general(self, parent):
+        """Add items to the parent summary table.
+        """
+        raise NotImplementedError()
+
+    def atropos_plots(self, parent):
+        """Add plots to the parent report.
+        """
+        raise NotImplementedError()
+
+
+class TrimModule(Submodule):
+    name = 'Atropos: Trimming'
+    anchor = 'atropos'
+    phase = 'trim'
+    info = ''
+
+    def __init__(self):
+        super(TrimModule, self).__init__()
+        self.atropos_trim_data = OrderedDict()
+        self.trim_sections = [
+            TrimmedLength()
+        ]
+
+    def add_data(self, sample_id, data):
+        super(TrimModule, self).add_data(sample_id, data)
+        self.atropos_general_data[sample_id] = data['derived'].copy()
+        self.atropos_general_data[sample_id]['mean_sequence_length'] = \
+            mean(data['derived']['mean_sequence_lengths'])
+        self.atropos_general_data[sample_id].update(dict(
+            (key, data[key])
+            for key in (
+                'total_record_count',
+                'total_bp_counts')))
+        
+        if 'trim' in data:
+            self.atropos_trim_data[sample_id] = data['trim']
+            self.atropos_general_data[sample_id]['fraction_bp_trimmed'] = \
+                1.0 - data['trim']['formatters']['fraction_total_bp_written']
+            for mod_dict in data['trim']['modifiers'].values():
+                if 'fraction_records_with_adapters' in mod_dict:
+                    self.atropos_general_data[sample_id]['fraction_records_with_adapters'] = \
+                        mean(mod_dict['fraction_records_with_adapters'])
+                    break
+
+    def atropos_assets(self, parent):
+        parent.css['atropos-assets/css/multiqc_atropos.css'] = os.path.join(
+            os.path.dirname(__file__), 'assets', 'css', 'multiqc_atropos.css')
+        parent.js['atropos-assets/js/multiqc_atropos.js'] = os.path.join(
+            os.path.dirname(__file__), 'assets', 'js', 'multiqc_atropos.js')
+
+    def atropos_general(self, parent):
+        """Add some single-number stats to the basic statistics table at the
+        top of the report.
+        """
+        headers = OrderedDict()
+        headers['input_format'] = {
+            'title': 'Format',
+            'description': 'Input File Format'
+        }
+        headers['total_record_count'] = {
+            'title': 'M Seqs',
+            'description': 'Total Sequences (millions)',
+            'min': 0,
+            'scale': 'Blues',
+            'modify': lambda x: x / 1000000,
```

Comment:
Normally I asked people to use `config.read_count_multiplier` here instead, so that the user can customise this for low / high sequencing depths. (See example in the [docs](http://multiqc.info/docs/#creating-a-table)).

However, since #607 by @vladsaveliev was merged the other day, these are now the defaults when using `shared_key = read_count` or `base_count`. So please just delete this lines.

Note that you do still need to use `config.read_count_prefix` and `config.read_count_desc`(or base pair equivalents) in the title and description. I'm still not sure on whether we should prefix / suffix these by default or not. Maybe we should..